ra/ra_test.go

```diff
@@ -2435,18 +2490,58 @@ func (sa *mockSAUnsafeAuthzReuse) GetAuthorizations(
 			},
 		},
 	}
-	// We can't easily access sa.authzMapToPB so we "inline" it for the mock :-)
-	resp := &sapb.Authorizations{}
-	for k, v := range authzs {
-		authzPB, err := sagrpc.AuthzToPB(*v)
-		if err != nil {
-			return nil, err
-		}
-		// Make a copy of k because it will be reassigned with each loop.
-		kCopy := k
-		resp.Authz = append(resp.Authz, &sapb.Authorizations_MapElement{Domain: &kCopy, Authz: authzPB})
+	return sa.AuthzMapToPB(authzs)
+}
+
+func (msa *mockSAUnsafeAuthzReuse) GetAuthorizations2(
```

Comment:
Here you mostly-duplicate the `GetAuthorizations` method that already exists on the mock. But above for a similar case, you modified the mock to contain a map, and return different transformations of the map for the v1 vs v2 authz case. That's a nice approach; we should do it here too.

Reply:
It's a bit complicated to do the same thing here as we need different IDs in these authorizations, whereas in the other example we are just using pared down authorizations that are the same for both methods.